### ISSUE TITLE
Transport Level Error Handling

### ISSUE THREAD

**Issue by TharmiganK** [ballerina-platform/ballerina-library#2866]
**Description:**

> Related to : [`Proposal: Error Handling #2669`](https://github.com/ballerina-platform/ballerina-standard-library/issues/2669)

Created this task to track transport level errors

**Describe your task(s)**

- Check the feasibility of handling transport level errors at error interceptors
- Move possible transport level validations such as header length and URI length validations to Request Interceptors and if the validation fails inject those errors to the pipeline so that they can be handled accordingly

**Comment by TharmiganK** [comment]
Tracking Table :
<table>
<thead>
  <tr>
    <th>Transport Level Error</th>
    <th>Is Interceptable</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Remote client closed the connection</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td> Remote host closed the connection</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>Connection between remote client and host is closed</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>Connection between client and remote host is closed</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>Promised stream is already rejected or stream is no longer valid</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>Could not obtain a connection within maximum wait time</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>Inbound response message already received</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>502 - Could not resolve host</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>502 - SSL connection failed</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>502 - Remote host closed the connection while SSL handshake</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>413 - Request entity too large</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>414 - Request uri too large</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>431 - Request Header Fields Too Large</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>Idle timeout triggerred errors</td>
    <td align="center">&#10060</td>
  </tr>
  <tr>
    <td>505 - HTTP version not supported</td>
    <td align="center">&#10060</td>
  </tr>
</tbody>
</table>

**Comment by shafreenAnfar** [comment]
Only these are valid right ?

Transport Level Error | Is Interceptable
-- | --
413 - Request entity too large | ❌
414 - Request uri too large | ❌
431 - Request Header Fields Too Large | ❌
505 - HTTP version not supported| ❌

**Comment by TharmiganK** [comment]
> Only these are valid right ?
> 
> Transport Level Error	Is Interceptable
> 413 - Request entity too large	❌
> 414 - Request uri too large	❌
> 431 - Request Header Fields Too Large	❌
> 505 - HTTP version not supported	❌

Yes, I have listed all the errors in the table to check whether they can be handled and will update the table once the feasibility is checked. Also, we may also able to intercept the idle timeout error occurs during the resource execution.
